### ISSUE TITLE
Custom validators need to specify a fixed field name for errors; args for custom validators?

### ISSUE THREAD

**Issue by chmielot** [biblius/validify#25]
Hi, I found your Reddit post about your new validify create by looking for solutions to filter and modify data before validating it. Your crate seems to be very promising.
I'm coming from the Validate crate and started to play around with yours by migrating my existing code. The first thing I have noticed is that for custom validators you have to return a fixed field name:

```
Err(ValidationError::new_field("field", 
```

So custom validators are actually not reusable and we see this in the examples also where the `greater_than_now` validator is used for two fields but only comes back as "field". The Validator crate was using the field name automatically. 
Am I missing something?

Second question: Is it possible to provide arguments to custom validators similar to `validate_args`?

**Comment by biblius** [comment]
Hi! Thanks for the feedback, glad you like the crate!

1. This is a valid point I've overlooked. The example is a bit outdated (my bad), as I've used it in the initial prototyping of the crate. Reusing custom validators will be one of the priorities in the next update when I get around to it, as well as updating that example to reflect the change. Nice catch!
2. As of now, no. I've considered adding this functionality, but personally I haven't found a use case that would justify the added complexity of such a feature. For example, how would we handle structs whose fields have multiple custom validators with different sets of arguments?  We could constrain the derive macro to work with only one type of arg for the whole struct (this would be the least painful in terms of the proc macro and ergonomics). We could also make it work on a per field basis, e.g. run each custom validation manually after the initial `validify` call, but at that point it is the same as rolling out your own, which I personally think is the way to go in such niche cases. That said, this opinion is not set in stone, and if we ever think up of a use case where it would be really beneficial, I'm open to adding it! :)
